src/aks-preview/azext_aks_preview/_helpers.py

```diff
@@ -321,3 +321,12 @@ def setup_common_guardrails_profile(level, version, excludedNamespaces, mc: Mana
             excludedNamespaces, enable_strip=True, keep_none=True, default_value=[])
 
     return mc
+
+
+def process_message(message):
+    result = message.split("\n")
+    if result[-2] != "[stderr]":
```

Comment:
Is there any way with less stringent conditions? BTW, could you leave a comment about the schema of the content?

Reply:
I haven't found a way to make it less stringent so feel free to suggest if you can think of a better way. This command utilizes the existing command of corresponding vm set type `az vmss run-command invoke` or `az vm run-command invoke`. The schema of the response is [here](https://learn.microsoft.com/en-us/python/api/azure-mgmt-compute/azure.mgmt.compute.v2019_03_01.models.instanceviewstatus?view=azure-python). The issue is `status` and `level` only shows the status of the `run-command` itself. Whether the script we invoke succeeded or not, we need to check the `message` to see if `[stderr]` contains anything. 

If you look at the example below, the first one is a success while the second one is a failure but the only way to tell is to look at the `message`.
![image](https://github.com/Azure/azure-cli-extensions/assets/49544763/50eb4e10-a383-4ea7-a538-315a9f7bca40)

Reply:
Since the message format is relatively flexible, I think the current processing method may not be robust enough. Not sure if [stdout/stderr] might contain another [stdout/stderr] or not.

Reply:
It won't since I control what the triggered script will output. I don't redirect any stderr so only when there's error triggering the script will it be logged to after the `[stderr]` part.